src/package-linker.js

```diff
@@ -296,7 +296,17 @@ export default class PackageLinker {
     // remove all extraneous files that weren't in the tree
     for (const loc of possibleExtraneous) {
       this.reporter.verbose(this.reporter.lang('verboseFileRemoveExtraneous', loc));
-      await fs.unlink(loc);
+
```

Comment:
This way, the `.has()` checks wouldn't have to change. As far as I see, the `files` set doesn't look used apart from those checks, so maybe it's enough?